docs/checkit.js

```diff
@@ -685,30 +685,43 @@ CheckIt.prototype.deleteCalendar = function() {
     
     // Guard against accidental clicks of the delete button
     
+
     var confirmation = confirm("Are you sure you want to delete your calendar?");
-    if (confirmation) {
-        
-        var checkit = this;
-                
-        return checkit.store.getActive().then(function(currentCalendarId) {
+    if (!confirmation) {
+      return Promise.reject("User cancelled the delete operation.");
+    }
+
+    var checkit = this;
+
+    return checkit.store.getActive()
+        .catch(function(err) {
+            console.error("Couldn't get active calendar id: " + err);
+        })
+        .then(function(currentCalendarId) {
             // remove calendar from dropdown
             checkit.removeFromCalendarDropdown(currentCalendarId,
-            checkit.$calendarDropdown);
+                                               checkit.$calendarDropdown);
+
             // remove calendar state from storage
-            return checkit.store.removeById(currentCalendarId).catch(function(err) {
-                console.log("Unable to remove calendar from storage.");
-                return err;
+            return checkit.store.removeById(currentCalendarId)
+                .catch(function(err) {
+                    console.error("Unable to remove calendar from storage.");
+                    // TODO: yes this is an error, but what do we want to do now?
+                    // Do we want to plow onward and clear the div or is this worth
+                    // stopping the entire app over?
```

Comment:
this .catch returns a rejected promise, which means the following
.then function does not get run.
 if we get this error, we could potentially handle it in the
 following .then function but we'd need to return a normal value
or a resolved promise...
                    
  so, I guess, if this is run because we are unable to remove a calendar
 from storage, then... we would still want the .then to run to remove 
the html of the calendar right?
 If we get to this catch function then there is something really wrong 
with the code, maybe it's better that the website just not work instead of
plowing on so that we know that something is seriously wrong? the way it's written now it's not plowing on